common-util/Details/ServiceState/index.jsx

```diff
@@ -106,26 +106,29 @@ export const ServiceState = ({
 
   /* ----- step 2 ----- */
   const handleStep2RegisterAgents = async () => {
-    const ids = [];
-    const instances = dataSource.map(
-      ({ agentAddresses, agentId, availableSlots }) => {
-        /**
-         * constructs agentIds:
-         * If there are 2 slots then agentInstances would need 2 addresses of instances
-         * ie. ids = [1, 1]
-         */
-        for (let i = 0; i < availableSlots; i += 1) {
-          ids.push(agentId);
-        }
-
-        return (agentAddresses || '').trim();
-      },
-    );
-    const agentInstances = (instances || [])
+    const trimArray = (string) => (string || [])
       .join()
       .split(',')
       .map((e) => e.trim());
 
+    const ids = [];
+    const instances = dataSource.map(({ agentAddresses, agentId }) => {
+      /**
+       * constructs agentIds:
+       * If there are 2 addresses of instances, then the agentIds will be [1, 1]
+       * example: agentAddresses = ['0x123', '0x456']
+       * agentId = 1
+       * ids = [1, 1]
+       */
+      const address = (agentAddresses || '').trim();
+      for (let i = 0; i < trimArray([address]).length; i += 1) {
+        ids.push(agentId);
+      }
+
+      return address;
+    });
+    const agentInstances = trimArray(instances || []);
+
```

Comment:
Will this account for if there are more than one `agentId`, meaning different agent Ids and different number of slots for each of them? What if the user puts more agent instances than the number of available slots?

Reply:
ok, the latter is something to observe with follow-up suggestions from Jose, we might want to check the user input for more errors than it is now.